src/gui/qgsextentgroupbox.cpp

```diff
@@ -238,6 +264,29 @@ void QgsExtentGroupBox::setOutputExtentFromLayer( const QgsMapLayer *layer )
   setOutputExtent( layer->extent(), layer->crs(), ProjectLayerExtent );
 }
 
+void QgsExtentGroupBox::setOutputExtentFromDrawOnCanvas()
+{
+  if ( mCanvas )
+  {
+    mMapToolPrevious = mCanvas->mapTool();
+    if ( !mMapToolExtent )
+    {
+      mMapToolExtent = new QgsMapToolExtent( mCanvas );
+      connect( mMapToolExtent, &QgsMapToolExtent::extentChanged, this, &QgsExtentGroupBox::extentDrawn );
+    }
+    mCanvas->setMapTool( mMapToolExtent );
+    window()->setVisible( false );
```

Comment:
I wonder if you'll need to activate/focus the canvas' window here too.

Something horrible like

```
  show();
  raise();
  setWindowState( windowState() & ~Qt::WindowMinimized );
  activateWindow();
```